server/game/core/gameSteps/prompts/HandlerMenuMultipleSelectionPrompt.js

```diff
@@ -0,0 +1,68 @@
+const HandlerMenuPrompt = require('./HandlerMenuPrompt.js');
+
+/**
+ * General purpose menu prompt that allows the user to submit multiple choices at once.
+ * Takes a choices object with menu options and an overall handler that is called when the user is done choosing.
+ * Handlers should return true in order to complete the prompt.
+ *
+ * The properties option object may contain the following:
+ * choices            - an array of titles for menu buttons
+ * maxSelected        - the maximum number of choices the player is allowed to submit. Leave undefined to allow the player to submit any number.
+ * handler            - a handler function that is called with the results of the prompt(which choices were selected) after the user clicks 'Done'
+ * activePromptTitle  - the title that should be used in the prompt for the choosing player.
+ * waitingPromptTitle - the title to display for opponents.
+ * source             - what is at the origin of the user prompt, usually a card; used to provide a default waitingPromptTitle, if missing
+ */
+class HandlerMenuMultipleSelectionPrompt extends HandlerMenuPrompt {
```

Comment:
This is great!! Very useful, nice work putting it together so fast.

One note for long-term is that we may eventually remove some of this logic and let the FE manage it as like checkboxes or something, in which case we'll just do what we do with e.g. Overwhelming Barrage where we provide the context and then just accept a chunk of data back from the FE indicating the user selection (as opposed to providing buttons and managing click-by-click in the BE). But for first release we want to minimize complexity in the FE so this is perfect for what we need for launch.